csFastFloat/Structures/DigitsBuffer.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Diagnostics;
+
+namespace csFastFloat.Structures
+{
+  internal unsafe struct DigitsBuffer
+  {
+    private fixed byte digits[(int)Constants.max_digits];
```

Comment:
This is C#'s _fixed sized buffer_ that can be accessed withou bound-checks.
But it's stack-allocated with 768 bytes. I'm not sure if this can result in any problems if other methods in the call-stack already rely on heavy stack-allocation. I assume for standard machines this won't be a problem, and as the discussion over in .NET-repo for integrating this algorithm looks like for Wasm / Blazor this may not be used in favor of a size-opt variant.

Alternatively we can heap allocate in the POH (pinned object heap, as special segment from the GC that pre-pins objects), so we could expose the pointer from here.
But that's an heap-allocation which should be avoided if possible, as there's a cost at collecting the "garbage".

Reply:
@gfoidl If I understand correctly, you replaced `digits = new max_digits[...]` with a stack-allocated struct, using the fact that max_digits is a compile-time constant. Of course, that's how the C++ equivalent is done.

This is a bit like the readonly byte span that were introduced elsewhere... that is, it is a way to avoid heap allocations.

I do hope that it should be fine. This makes the whole thing "never allocating" and you can compute exactly how much memory you do use (in the worst case).

I am not sure why that would be a problem with WASM... With WASM, if you know precisely the memory usage (as a fixed quantity), you are golden. Aren't you?

Reply:
You understood correct!

> With WASM, if you know precisely the memory usage

But I don't know WASM, so that's my problem 😉 (expect form wikipedia).

Reply:
could you make the indexers simpler by using something like::
```cs
public ref byte this[uint index]
{
      get
      {
        Debug.Assert(index < Constants.max_digits);
        return ref digits[index];
      }
}
```

Reply:
I thought about this, but didn't follow that approach and didn't measure it.
My fear was that we may hit some JIT-limits, if the `ref byte` will be used in some non-direct ways like casting. That's why the canonical way was gone.

But maybe my fear is without merit -- feel free to validate it.